src/pages/courts/index.tsx

```diff
@@ -60,10 +67,12 @@ const courts = () => {
         <Table variant="simple">
           <TableHeader />
           <Tbody>
-            {courtsData.map((court) => {
-              court["createdAt"] = formatDate(court.createdAt);
-              court["updatedAt"] = formatDate(court.updatedAt);
-              const { _id, name, createdAt, updatedAt } = court;
+            {isLoading && <Text>Please wait while the courts data are loading...</Text>}
+            {courtsData?.map((court: ICourt) => {
+              const courtCopy = { ...court };
+              courtCopy["createdAt"] = formatDate(court.createdAt);
+              courtCopy["updatedAt"] = formatDate(court.updatedAt);
+              const { _id, name, createdAt, updatedAt } = courtCopy;
```

Comment:
As the createdAt and updatedAt field are read-only defined at the backend. We do need to do can deep copy of the court data so that the createdAt and updatedAt date format can be transformed into human-read-friendly,